src/include/catalog/catalog.h

```diff
@@ -238,14 +236,19 @@ class Catalog {
     inline function::ScalarMacroFunction* getScalarMacroFunction(std::string name) const {
         return catalogContentForReadOnlyTrx->macros.at(name).get();
     }
+    inline storage::BMFileHandle* getNodeGroupsMetaFH() const { return nodeGroupsMetaFH.get(); }
 
 private:
     inline bool hasUpdates() { return catalogContentForWriteTrx != nullptr; }
 
+    void addMetaDAHeaderPageForProperty(
+        const common::LogicalType& dataType, MetaDiskArrayHeaderInfo& diskArrayHeaderInfo);
+
 protected:
     std::unique_ptr<CatalogContent> catalogContentForReadOnlyTrx;
     std::unique_ptr<CatalogContent> catalogContentForWriteTrx;
     storage::WAL* wal;
+    std::unique_ptr<storage::BMFileHandle> nodeGroupsMetaFH;
```

Comment:
Moved to storage manager, though i'm not confident which is the best place. I believe it will become self-evident as we move on.

Reply:
Why? Catalog clearly is not a place to keep track of disk-related fields, such as filehandles. In addition this field already exists (as raw pointer) in node_table.h, which makes sense.

Reply:
My understanding was regardless logical  (schema) or physical (metadata), these information are metadata info around physically stored table data. It depends on if we should separate logical and physical or not. I was a bit indecisive, but now I think separating them makes more sense. So I will make the change.

I tried to move metadataFH to StorageManager. The part I like is that it avoids involving Catalog to interact with metadata file, while the annoying part is that during wal replaying, when it comes to recovery, because there is no StorageManager present, we need to somehow (re)construct a `metadataFH`, which i don't think it's a correct design.

Following this, not related to this PR, I've been quite confused why we chose to do recovery before we construct Catalog and StorageManager objects. (I kinda cannot remember why we have to do that) Is this still a must-to-go design now?